src/Symfony/Component/Translation/Translator.php

```diff
@@ -192,17 +204,17 @@ public function trans($id, array $parameters = array(), $domain = null, $locale
             $domain = 'messages';
         }
 
-        return strtr($this->getCatalogue($locale)->get((string) $id, $domain), $parameters);
+        return $this->formatter->format($this->getCatalogue($locale)->get((string) $id, $domain), $locale, $parameters);
     }
 
     /**
      * {@inheritdoc}
      */
     public function transChoice($id, $number, array $parameters = array(), $domain = null, $locale = null)
     {
-        $parameters = array_merge(array(
-            '%count%' => $number,
-        ), $parameters);
+        if (!$this->formatter instanceof ChoiceMessageFormatterInterface) {
+            throw new InvalidArgumentException(sprintf('The formatter "%s" does not support plural translations.', get_class($this->formatter)));
```

Comment:
The formatter is not an argument here, what about throwing a LogicException instead?